aws-lc-rs/src/key_wrap.rs

```diff
@@ -0,0 +1,328 @@
+        let mut aes_key = MaybeUninit::<AES_KEY>::uninit();
+
+        let key_bits: u32 = (key.len() * 8).try_into().map_err(|_| Unspecified)?;
+
+        if 0 != unsafe { AES_set_encrypt_key(key.as_ptr(), key_bits, aes_key.as_mut_ptr()) } {
+            return Err(Unspecified);
+        }
+
+        let aes_key = unsafe { aes_key.assume_init() };
+
+        let out_len = match self {
+            WrappingMode::Padded => {
+                let mut out_len = MaybeUninit::<usize>::uninit();
+
+                if 1 != indicator_check!(unsafe {
+                    AES_wrap_key_padded(
+                        &aes_key,
+                        output.as_mut_ptr(),
+                        out_len.as_mut_ptr(),
+                        output.len(),
+                        input.as_ptr(),
+                        input.len(),
+                    )
+                }) {
+                    return Err(Unspecified);
+                }
+
+                unsafe { out_len.assume_init() }
+            }
+            WrappingMode::Unpadded => {
+                let out_len = indicator_check!(unsafe {
+                    AES_wrap_key(
+                        &aes_key,
+                        null(),
+                        output.as_mut_ptr(),
+                        input.as_ptr(),
+                        input.len(),
+                    )
+                });
+
+                if out_len == -1 {
+                    return Err(Unspecified);
+                }
+
+                let out_len: usize = out_len.try_into().map_err(|_| Unspecified)?;
+
+                out_len
+            }
+        };
+
+        Ok(&mut output[..out_len])
+    }
+
+    fn unwrap<'output>(
+        &self,
+        key: &[u8],
+        input: &[u8],
+        output: &'output mut [u8],
+    ) -> Result<&'output mut [u8], Unspecified> {
+        let mut aes_key = MaybeUninit::<AES_KEY>::uninit();
+
+        let key_bits: u32 = (key.len() * 8).try_into().map_err(|_| Unspecified)?;
+
+        if 0 != unsafe { AES_set_decrypt_key(key.as_ptr(), key_bits, aes_key.as_mut_ptr()) } {
+            return Err(Unspecified);
+        }
+
+        let aes_key = unsafe { aes_key.assume_init() };
+
+        let out_len = match self {
+            WrappingMode::Padded => {
+                let mut out_len = MaybeUninit::<usize>::uninit();
+
+                if 1 != indicator_check!(unsafe {
+                    AES_unwrap_key_padded(
+                        &aes_key,
+                        output.as_mut_ptr(),
+                        out_len.as_mut_ptr(),
+                        output.len(),
+                        input.as_ptr(),
+                        input.len(),
+                    )
+                }) {
+                    return Err(Unspecified);
+                };
+
+                unsafe { out_len.assume_init() }
+            }
+            WrappingMode::Unpadded => {
+                let out_len = indicator_check!(unsafe {
```

Comment:
Since we expect the return value to be `|in_len| - 8`, we could have a debug assertion on that as a sanity check.